espressif/vscode-esp-idf-extension
# [VSC-1546] fix version switch by persist sys python in idfSetup

## Description

Because ESP-IDF version now computes the virtual env python path from a `idf.pythonInstallPath` and current ESP-IDF version, there is an issue for v4.4.8 (uses embed Python 3.8) and v5.3.1 (use embed Python 3.11) calculates the wrong env path because this embed python is not persisted in idfSetup in extension global context. This leads to a wrong python env path calculated which failed the verification when showing the list of available IDF setups in the version switcher or `ESP-IDF: Configure ESP-IDF extension` command setup wizard.

Fixes #1378 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Steps to test this pull request

1. Delete current global context IDF Setups by running the `ESP-IDF: Clear ESP-IDF Saved Setups` command.
2. Use the `ESP-IDF: Configure ESP-IDF extension` to find existing ESP-IDF setup for v5.x.x. This will add this setup to the global extension context.
3. Use the `ESP-IDF: Configure ESP-IDF extension` to find existing ESP-IDF setup for v4.4.x. This will add this setup to the global extension context.
4. Click on "ESP-IDF: Configure ESP-IDF extension". See that both ESP-IDF v5.3.1 and v4.4.8 are being shown in the list of existing setups.

- Expected behaviour:

See that both ESP-IDF v5.3.1 and v4.4.8 are being shown in the list of existing setups in `ESP-IDF: Configure ESP-IDF extension` command setup wizard and status bar version switcher.

- Expected output:

See that both ESP-IDF v5.3.1 and v4.4.8 are being shown in the list of existing setups in `ESP-IDF: Configure ESP-IDF extension` command setup wizard and status bar version switcher.

## How has this been tested?

Manual testing as described above.

**Test Configuration**:
* ESP-IDF Version: 5.3.1 and v4.4.8
* OS (Windows,Linux and macOS): Windows (this was not an issue in MacOS or Linux as system python is the same among esp-idf versions)

## Checklist
